src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp

```diff
@@ -256,6 +256,10 @@ bool ov::pass::MOCTransformations::run_on_model(const std::shared_ptr<ngraph::Fu
             param->set_partial_shape(input_shapes.at(param.get()));
         }
         f->validate_nodes_and_infer_types();
+    } else {
```

Comment:
could you add a comment explaining why do we need it here?

Reply:
Removed it.